src/main/java/optimus/Parser.java

```diff
@@ -0,0 +1,104 @@
+package optimus;
+import java.io.IOException;
+import java.util.List;
+
+public class Parser {
+
+    public static Task parseCommand(String text, TaskList record, Ui ui, Storage storage) throws OptimusException, IOException {
+        text = text.trim();
+        if (text.equals("bye")) {
+            storage.saveToFile(record.getTasks());
+            ui.showGoodbye();
+            System.exit(0);
+        } else if (text.startsWith("find ")) {
+            String keyword = text.substring(5).trim();
+            if (keyword.isEmpty()) {
+                throw new OptimusException("The keyword provided is empty");
+            }
+            record.findTasks(keyword, ui);
+        } else if (text.equals("list")) {
+            ui.listTasks(record);
```

Comment:
I like how the code is quite modular and broken down into the different functions! 👍